cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/template/etl/api/Transformation.java

```diff
@@ -23,9 +23,10 @@
  *
  * @param <IN> Type of input object
  * @param <OUT> Type of output object
+ * @param <T> execution context
  */
 @Beta
-public interface Transformation<IN, OUT> {
+public interface Transformation<IN, OUT, T extends TransformContext> extends StageLifecycle<T> {
```

Comment:
Why `Transformation` always need lifecycle? Seems like it should be `Transform` who implements lifecycle.